code/modules/mob/living/simple_animal/friendly/mule_bot/mule_bot_modules.dm

```diff
@@ -0,0 +1,182 @@
+	max_storage_space = 12
+	storage_slots = null
+
+/*
+Personal storage module
+
+based on keycard you get your own private storage, imagine MC enderchests sorta but not shared with other bots
+
+*/
+
+/obj/item/mule_module/personal_storage
+	name = "Personal storage module"
+	desc = "A module that allows the mule to carry various items for various individuels"
+	overlay_icon_state = "marine_rocket_full"
+	overlay_icon = 'icons/mob/kerfus.dmi'
+	overlay_icon_state = "backpack"
+	module_desc = "The installed module allows you to have a personal storage inside the mule based on ID"
+	var/list/obj/item/storage/mule_pack/small/packs = list()
+
+/obj/item/mule_module/personal_storage/apply(mob/living/simple_animal/mule_bot/mule)
+	RegisterSignal(mule,COMSIG_ATOM_ATTACK_HAND, PROC_REF(acces_storage))
+	return ..()
+
+/obj/item/mule_module/personal_storage/unapply(delete_mod = TRUE)
+	UnregisterSignal(attached_mule, COMSIG_ATOM_ATTACK_HAND)
+	return ..()
+
+//Since every person gets there own storage bit with this module, we need to either create new storage for the person or let it acces the old one
+/obj/item/mule_module/personal_storage/proc/acces_storage(mob/mule, mob/user)
+	var/card = user.get_idcard(TRUE)
+	if(!card)
+		to_chat(user,span_warning("You need an valid ID to store items here!"))
+		return
+	if(!packs[card])
+		packs[card] = new /obj/item/storage/mule_pack/small(src)
+	packs[card].open(user)
+
+/*
+Spot light module
+
+Bassicly baldur jeager mod for robot. gives light
+
+*/
+
+/obj/item/mule_module/light
+	name = "Spot light module"
+	desc = "This module lets the bot cast a bright light."
+	var/mod_Light_power = 10
+	var/mod_Light_range = 23
+	var/mod_light_color = "#f0e0b6"
+
+/obj/item/mule_module/light/apply(mob/living/simple_animal/mule_bot/mule)
+	mule.set_light_range_power_color(mod_Light_power,mod_Light_range,mod_light_color)
+	mule.set_light_on(TRUE)
+	return ..()
+
+/obj/item/mule_module/light/unapply(delete_mod)
+	attached_mule.set_light_range_power_color(initial(attached_mule.light_range) ,initial(attached_mule.light_power),initial(attached_mule.light_color))
+	attached_mule.set_light_on(FALSE)
+	return ..()
+
+/*
+Flare placing module
+
+Place flares on turf if we find its too dark
+
+
+*/
+
+/obj/item/mule_module/flare_placer
+	name = "Flare placing module"
+	desc = "This module places flare automaticly when its too dark."
+	COOLDOWN_DECLARE(flare_place)
+
+/obj/item/mule_module/flare_placer/apply(mob/living/simple_animal/mule_bot/mule)
+	RegisterSignal(mule, COMSIG_MOVABLE_MOVED, PROC_REF(check_lumcount))
+	return ..()
+
+
+/obj/item/mule_module/flare_placer/proc/check_lumcount(atom/bot)
+	SIGNAL_HANDLER
+	if(!COOLDOWN_CHECK(src,flare_place))
+		return
+	if(!isturf(bot.loc))
+		return
+	var/turf/T = bot.loc
+	if(T.dynamic_lumcount < 0.5 && T.luminosity < 1)
+		var/obj/item/explosive/grenade/flare/F = new(T)
+		F.activate(bot)
+		//lum count wont update properly since lighting uses moved signals
+		F.forceMove(T)
```

Comment:
I checked and this is handled on toggling as well so ???

Reply:
This is what happens if I uncomment this and set the cooldown to 0. lum count wont update

https://cdn.discordapp.com/attachments/500801682058379265/1096770632819150908/dreamseeker_SwT8gM76IF.mp4